.gitignore

```diff
@@ -0,0 +1,2 @@
+# Ignore the common build directory
+build/
```

Comment:
Do you think this should be our recommended layout? Yes, the CI sets things up like this. But I myself prefer a proper out-of-source build.

I was expecting that you would be adding the files to be ignored in an in-source build. That would be useful. Though I have strongly opposed feelings against in-source builds as well...

Reply:
Personally I have no particular preference for in-source builds (both kinds), or out-of-source builds; `git clean` takes care of any excess files I may want to get rid of. In my experience in-source builds are somewhat easier for people not experienced with CMake, but the difference is not at all huge.

Reply:
Outside the ATLAS context, I've never had a problem building anything with a `build` folder inside the git repo.